src/tools.rs

```diff
@@ -0,0 +1,79 @@
+
+pub trait Slice {

```

Comment:
That piece of code is quite interesting, thanks! Although the functions there should still be marked unsafe, the Copy trait is not quite enough to ensure safety. Hence the unsafe trait I added to my crate (which, incidentally, would be very easy to automatically derive in compiler).